lib/trino-orc/src/main/java/io/trino/orc/OrcReaderOptions.java

```diff
@@ -111,111 +117,171 @@ public boolean isNestedLazy()
     {
-        return new OrcReaderOptions(
-                bloomFiltersEnabled,
-                maxMergeDistance,
-                maxBufferSize,
-                tinyStripeThreshold,
-                streamBufferSize,
-                maxBlockSize,
-                lazyReadSmallRanges,
-                nestedLazy);
+        return new Builder(this)
+                .withStreamBufferSize(streamBufferSize)
+                .build();
     }
 
     public OrcReaderOptions withMaxReadBlockSize(DataSize maxBlockSize)
     {
-        return new OrcReaderOptions(
-                bloomFiltersEnabled,
-                maxMergeDistance,
-                maxBufferSize,
-                tinyStripeThreshold,
-                streamBufferSize,
-                maxBlockSize,
-                lazyReadSmallRanges,
-                nestedLazy);
+        return new Builder(this)
+                .withMaxBlockSize(maxBlockSize)
+                .build();
     }
 
     // TODO remove config option once efficacy is proven
     @Deprecated
     public OrcReaderOptions withLazyReadSmallRanges(boolean lazyReadSmallRanges)
     {
-        return new OrcReaderOptions(
-                bloomFiltersEnabled,
-                maxMergeDistance,
-                maxBufferSize,
-                tinyStripeThreshold,
-                streamBufferSize,
-                maxBlockSize,
-                lazyReadSmallRanges,
-                nestedLazy);
+        return new Builder(this)
+                .withLazyReadSmallRanges(lazyReadSmallRanges)
+                .build();
     }
 
     // TODO remove config option once efficacy is proven
     @Deprecated
     public OrcReaderOptions withNestedLazy(boolean nestedLazy)
     {
-        return new OrcReaderOptions(
-                bloomFiltersEnabled,
-                maxMergeDistance,
-                maxBufferSize,
-                tinyStripeThreshold,
-                streamBufferSize,
-                maxBlockSize,
-                lazyReadSmallRanges,
-                nestedLazy);
+        return new Builder(this)
+                .withNestedLazy(nestedLazy)
+                .build();
+    }
+
+    @Deprecated
+    public OrcReaderOptions withShortZoneIdEnabled(boolean shortZoneIdEnabled)
+    {
+        return new Builder(this)
+                .withShortZoneIdEnabled(shortZoneIdEnabled)
+                .build();
+    }
+
+    private static class Builder
+    {
+        private boolean bloomFiltersEnabled;
+        private DataSize maxMergeDistance;
+        private DataSize maxBufferSize;
+        private DataSize tinyStripeThreshold;
+        private DataSize streamBufferSize;
+        private DataSize maxBlockSize;
+        private boolean lazyReadSmallRanges;
+        private boolean nestedLazy;
+        private boolean shortZoneIdEnabled;
+
+        private Builder(OrcReaderOptions orcReaderOptions)
+        {
+            requireNonNull(orcReaderOptions, "orcReaderOptions is null");
```

Comment:
I think it's not needed, you use it line bellow and in case of NPE Java will throw error message with line number

Reply:
Isn't it good to check for null,  rather than java throwing NPE?

Reply:
the reason why you see `rnn` in most places is to make sure error message is usable. Older JDK has useless NPE errors, since JDK 17 NPE on field access gives a useful error message making the `rnn` redundant (but only in cases where the field is immediately dereferenced, in all other cases we still try to use `rnn`).

Reply:
my personal opinion is to keep using `rnn` except for Airlift's config classes since we always deref configs in ctor (because they are mutable).